nipype/pipeline/plugins/multiproc.py

```diff
@@ -100,3 +177,126 @@ def _report_crash(self, node, result=None):
+        # The most resource consuming jobs run first
+        jobids = sorted(jobids,
+                        key=lambda item: (self.procs[item]._interface.estimated_memory_gb,
+                                          self.procs[item]._interface.num_threads))
+
+        logger.debug('Free memory (GB): %d, Free processors: %d',
+                     free_memory_gb, free_processors)
+
+        # While have enough memory and processors for first job
+        # Submit first job on the list
+        for jobid in jobids:
+            logger.debug('Next Job: %d, memory (GB): %d, threads: %d' \
+                         % (jobid, self.procs[jobid]._interface.estimated_memory_gb,
+                            self.procs[jobid]._interface.num_threads))
+
+            if self.procs[jobid]._interface.estimated_memory_gb <= free_memory_gb and \
+               self.procs[jobid]._interface.num_threads <= free_processors:
+                logger.info('Executing: %s ID: %d' %(self.procs[jobid]._id, jobid))
+                executing_now.append(self.procs[jobid])
+
+                if isinstance(self.procs[jobid], MapNode):
+                    try:
+                        num_subnodes = self.procs[jobid].num_subnodes()
+                    except Exception:
+                        self._clean_queue(jobid, graph)
+                        self.proc_pending[jobid] = False
+                        continue
+                    if num_subnodes > 1:
+                        submit = self._submit_mapnode(jobid)
+                        if not submit:
+                            continue
+
+                # change job status in appropriate queues
+                self.proc_done[jobid] = True
+                self.proc_pending[jobid] = True
+
+                free_memory_gb -= self.procs[jobid]._interface.estimated_memory_gb
+                free_processors -= self.procs[jobid]._interface.num_threads
+
+                # Send job to task manager and add to pending tasks
+                if self._status_callback:
+                    self._status_callback(self.procs[jobid], 'start')
+                if str2bool(self.procs[jobid].config['execution']['local_hash_check']):
+                    logger.debug('checking hash locally')
+                    try:
+                        hash_exists, _, _, _ = self.procs[
+                            jobid].hash_exists()
+                        logger.debug('Hash exists %s' % str(hash_exists))
+                        if (hash_exists and (self.procs[jobid].overwrite == False or \
+                                             (self.procs[jobid].overwrite == None and \
+                                              not self.procs[jobid]._interface.always_run))):
+                            self._task_finished_cb(jobid)
+                            self._remove_node_dirs()
+                            continue
+                    except Exception:
```

Comment:
I think sometimes it makes sense to be able to be prepared to catch any exception such that we can still clean and remove that node from the proc_pending queue. However, I am not sure what specific exception would come up here, could be an IOError from a missing filepath or perhaps a KeyError or AttributeError when trying to access various parameters of the procs dictionary. However, if you like, we can remove the try block altogether